modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/SAPAccessControlPolicy.java

```diff
@@ -184,6 +188,17 @@ private void _checkAccess(
 		if (allowedServiceSignatures.contains(StringPool.STAR) ||
 			allowedServiceSignatures.contains(className)) {
 
+			if (className.contains("HeadlessDiscoveryAPIApplication")) {
```

Comment:
This is a bit unusual because you are adding a very specific case into the SAPAccessControlPolicy which is otherwise entirely data driven. Did you explore a more generic approach and found it not to be possible?

Reply:
Hi @stian-sigvartsen 
Yes, in this case, it is a very specific scenario that occurs with only one class involved.